src/Features/CSharp/Portable/GenerateMember/GenerateVariable/CSharpGenerateVariableService.cs

```diff
@@ -69,7 +78,6 @@ protected override bool ContainingTypesOrSelfHasUnsafeKeyword(INamedTypeSymbol c
         {
             identifierToken = identifierName.Identifier;
             if (identifierToken.ValueText != string.Empty &&
-                !identifierName.IsVar &&
```

Comment:
This was blocking `var` when I originally wanted it to become available to be fixed. Even considering that I am going to change it, `var` will be filtered before executing this line, so there is no sense in this line anyway

Reply:
> var will be filtered before executing this line, 

What does that filtering?

Reply:
`TryInitializeIdentifierNameState` is only called in `TryInitializeSimpleName`. `TryInitializeSimpleName` is not called, when `IsIdentifierNameGeneration` returns false:
https://github.com/dotnet/roslyn/blob/ae66fbc92c8b717eb6a9eff5bb7b3965694e50e4/src/Features/Core/Portable/GenerateMember/GenerateVariable/AbstractGenerateVariableService.State.cs#L106-L117
`IsIdentifierNameGeneration` returns false for `var`, because `IsProbablySyntacticConstruct` will be true, and this value is inverted